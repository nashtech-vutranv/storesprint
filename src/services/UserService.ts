import {AxiosInstance} from 'axios'
import {IPagination} from '../interface/pagination'
import {IUsersFilter} from '../interface/filters'
import {
  generateFilterPaginationQuery,
} from '../helpers/queryParams'
import {IUser} from '../interface/user'
import {IUserSetting} from '../interface/userSetting'

class UserService {
  private axiosClient

  constructor(axiosClient: AxiosInstance) {
    axiosClient.defaults.baseURL = process.env.REACT_APP_API_SERVER_URL
    this.axiosClient = axiosClient
  }

  getUsersByFilter(
    filterPaginationQuery: Omit<IPagination, 'keyword'>,
    usersFilterObj: IUsersFilter
  ) {
    const requestUrl = `/users/filter?${generateFilterPaginationQuery(
      filterPaginationQuery
    )}`

    const usersData = this.axiosClient.post(requestUrl, usersFilterObj)
    return usersData
  }

  getUserById(userId: string | undefined) {
    const userData = this.axiosClient.get(`/users/${userId}`)
    return userData
  }

  addUser(userData: IUser) {
    return this.axiosClient.post('/users', userData)
  }

  editUser(userId: string, data: IUser) {
    return this.axiosClient.put(`/users/${userId}`, data)
  }

  getCurrentUserSetting() {
    return this.axiosClient.get('/users/current-user')
  }

  updateCurrentUserSetting(userSettingData: IUserSetting) {
    return this.axiosClient.put('/users/current-user/setting', userSettingData)
  }
}

export default UserService
