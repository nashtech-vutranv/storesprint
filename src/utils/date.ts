import moment from 'moment'
import {FORMAT_DATE, FORMATE_DATE_NO_HOURS} from '../constants'

export const formatDate = (date: string, format: string = FORMAT_DATE) => {
  return moment(date).format(format)
}

export const formatDateNoHours = (date: string, format: string = FORMATE_DATE_NO_HOURS) => {
  return moment(date).format(format)
}
