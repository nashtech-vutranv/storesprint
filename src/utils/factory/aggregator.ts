import faker from '@faker-js/faker'

export const mockAggregator = () => {
  return {
    id: faker.random.word(),
    name: faker.random.word(),
    version: faker.datatype.number(),
    status: faker.datatype.string(),
  }
}

export const mockAggregators = () => {
  return {
    data: {
      content: Array.apply(null, Array(10)).map(() => mockAggregator()),
      totalElements: 10,
    },
  }
}
