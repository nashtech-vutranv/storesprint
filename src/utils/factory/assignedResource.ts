import faker from '@faker-js/faker'

export const mockAssignedResource = () => {
  return {
    data: {
      createdAt: faker.datatype.string(),
      id: faker.datatype.string(),
      modifiedAt: faker.datatype.string(),
      organizationResponse: {
        additionalInformation: faker.datatype.string(),
        addressLine1: faker.datatype.string(),
        addressLine2: null,
        addressLine3: null,
        city: faker.datatype.string(),
        contactEmailAddress: faker.datatype.string(),
        contactName: faker.datatype.string(),
        contactPhoneNumber: faker.datatype.string(),
        createdAt: faker.datatype.string(),
        erpId: faker.datatype.string(),
        id: faker.datatype.string(),
        modifiedAt: faker.datatype.string(),
        name: faker.datatype.string(),
        postCode: faker.datatype.string(),
        registrationNumber: faker.datatype.string(),
        status: faker.datatype.string(),
        vatRegistrationNumber: faker.datatype.string(),
        version: faker.datatype.number(),
      },
      siteResponse: {
        createdAt: faker.datatype.string(),
        erpId: faker.datatype.string(),
        id: faker.datatype.string(),
        modifiedAt: faker.datatype.string(),
        name: faker.datatype.string(),
        organizationId: faker.datatype.string(),
        status: faker.datatype.string(),
        url: faker.datatype.string(),
        version: faker.datatype.number(),
      },
      userResponse: null,
      version: faker.datatype.number(),
    },
  }
}
