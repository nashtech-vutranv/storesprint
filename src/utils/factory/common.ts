import {IPaginationResponse} from '../../interface'

export const mockOption = () => ({
  value: '',
  label: '',
})

export const mockPaginationResponse = (): IPaginationResponse => {
  return {
    pageable: {
      sort: {
        empty: true,
        sorted: false,
        unsorted: true,
      },
      offset: 0,
      pageSize: 3,
      pageNumber: 0,
      unpaged: false,
      paged: true,
    },
    last: false,
    totalPages: 2,
    totalElements: 5,
    number: 0,
    size: 3,
    first: true,
    numberOfElements: 3,
    sort: {
      empty: true,
      sorted: false,
      unsorted: true,
    },
    empty: false,
  }
}
