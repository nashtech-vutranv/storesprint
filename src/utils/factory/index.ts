export * from './organization'
export * from './warehouse'
export * from './aggregator'
export * from './marketplaces'
export * from './currency'
export * from './relationshipAM'
export * from './marketplaceRelationships'
export * from './product'
export * from './order'
export * from './listingStatus'
export * from './common'
export * from './defaultPropertyValue'
export * from './mappings'
export * from './productProperties'
export * from './assignedResource'
export * from './productCategoryMappings'
export * from './productPropertyMappings'
export * from './courier'
export * from './valueMapping'
export * from './courierMappings'
export * from './returnsrefunds'
export * from './deliveryTypeMappings'
