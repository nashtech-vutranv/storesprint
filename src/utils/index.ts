export * from './array'
export * from './layout'
export * from './date'
export * from './conditionPromise'
