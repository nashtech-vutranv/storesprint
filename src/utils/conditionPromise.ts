import {mockOrganizations, mockSites, mockSiteLocales} from './factory/organization'
import {mockMarketplaces, mockCalculateAssignMarketplaces} from './factory/marketplaces'
import {mockAssignedProducts, mockAssignedProductsResponse} from './factory/product'
import {mockWarehouses} from './factory/warehouse'
import {mockMarketplaceRelationship} from './factory/marketplaceRelationships'

type IMockType = 'organization' | 'site' | 'siteLocale' |
 'marketplace' | 'calculateMarketplace' | 'warehouse' 
 | 'marketplaceRelationship' | 'assignedProduct' | 'assignedProductResponse'

export const conditionPromise = (objectFilter: unknown, typeMock: IMockType, objectPaginationQuery?: any) => {
  if (objectFilter || (objectFilter && objectPaginationQuery)) {
    switch (typeMock) {
      case 'organization': return Promise.resolve(mockOrganizations())
      case 'site': return Promise.resolve(mockSites())
      case 'siteLocale': return Promise.resolve(mockSiteLocales())
      case 'marketplace': return Promise.resolve(mockMarketplaces())
      case 'calculateMarketplace': return Promise.resolve(mockCalculateAssignMarketplaces())
      case 'warehouse': return mockWarehouses()
      case 'marketplaceRelationship': return mockMarketplaceRelationship()
      case 'assignedProduct': return Promise.resolve(mockAssignedProducts())
      case 'assignedProductResponse': return Promise.resolve(mockAssignedProductsResponse())

      default: return {}
    }
  } else return Promise.reject()
}