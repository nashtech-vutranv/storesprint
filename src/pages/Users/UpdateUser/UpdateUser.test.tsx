import {render, waitFor} from '@testing-library/react'
import axios from 'axios'
import '@testing-library/jest-dom'
import {MemoryRouter} from 'react-router-dom'
import UserService from '../../../services/UserService'
import {mockOption} from '../../../utils/factory'
import {UpdateUser} from '../'

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useParams: jest.fn(() => ({userId: '1234'})),
}))

beforeEach(() => {
  window.history.pushState(
    {
      selectedOrganization: mockOption(),
      selectedSites: [mockOption()],
    },
    '',
    ''
  )
})

const mockData = {
  id: '',
  erpId: '',
  name: '',
  firstName: 'Matthew',
  lastName: 'Moulding',
  emailAddress: 'matthewmoulding@thg.com',
  createdAt: '',
  modifiedAt: '',
  version: 0,
  status: 'ACTIVE',
}

const tree = render(
    <MemoryRouter>
      <UpdateUser />
    </MemoryRouter>
  )

describe('Update User Page', () => {
  it('should match snapshot without data', async () => {
    await waitFor(() => {
      expect(tree).toMatchSnapshot()
    })
  })

  it('should match snapshot with data', async () => {
    jest
      .spyOn(new UserService(axios) as any, 'getUserById')
      .mockImplementation((id) => Promise.resolve(mockData))
    await waitFor(() => {
      expect(tree).toMatchSnapshot()
    })
    
  })
})
