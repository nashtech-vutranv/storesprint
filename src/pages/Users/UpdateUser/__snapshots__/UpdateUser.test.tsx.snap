// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Update User Page should match snapshot with data 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body />,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Update User Page should match snapshot without data 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="card-form mt-3 card"
      >
        <div
          class="card-header"
        >
          <h4
            class="card-form__title"
          >
            user_edit_page_title
          </h4>
        </div>
        <div
          class="card-body"
        >
          <div
            class="row"
          >
            <form
              autocomplete="on"
              class="form-layout"
            >
              <div
                class="container-fluid"
              >
                <div
                  class="align-items-center py-1 row"
                >
                  <div
                    class="col-4"
                  >
                    <label
                      for="id"
                    >
                      Id
                    </label>
                  </div>
                  <div
                    class="p-fluid col-8"
                  >
                    <input
                      aria-label="disabled"
                      class="p-inputtext p-component p-disabled p-1"
                      disabled=""
                      id="id"
                      name="id"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="align-items-center py-1 row"
                >
                  <div
                    class="col-4"
                  >
                    <label
                      for="erpId"
                    >
                      organization_details_erpId
                    </label>
                  </div>
                  <div
                    class="p-fluid col-8"
                  >
                    <input
                      aria-label="disabled"
                      class="p-inputtext p-component p-disabled w-full p-1"
                      disabled=""
                      id="erpId"
                      maxlength="16"
                      name="erpId"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="row"
                />
                <div
                  class="align-items-center py-1 row"
                >
                  <div
                    class="col-4"
                  >
                    <label
                      class="required"
                      for="firstName"
                    >
                      user_form_first_name
                    </label>
                  </div>
                  <div
                    class="p-fluid col-8"
                  >
                    <input
                      class="p-inputtext p-component w-full p-1"
                      id="firstName"
                      maxlength="128"
                      name="firstName"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="align-items-center py-1 row"
                >
                  <div
                    class="col-4"
                  >
                    <label
                      class="required"
                      for="lastName"
                    >
                      user_form_last_name
                    </label>
                  </div>
                  <div
                    class="p-fluid col-8"
                  >
                    <input
                      class="p-inputtext p-component w-full p-1"
                      id="lastName"
                      maxlength="128"
                      name="lastName"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="align-items-center py-1 row"
                >
                  <div
                    class="col-4"
                  >
                    <label
                      class="required"
                      for="contactEmailAddress"
                    >
                      user_form_email_address
                    </label>
                  </div>
                  <div
                    class="p-fluid col-8"
                  >
                    <input
                      aria-label="disabled"
                      class="p-inputtext p-component p-disabled w-full p-1"
                      disabled=""
                      id="emailAddress"
                      maxlength="256"
                      name="emailAddress"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="row"
                />
                <div
                  class="align-items-center py-1 row"
                >
                  <div
                    class="col-4"
                  >
                    <label
                      class=""
                      for="isActive"
                    >
                      organization_details_status_active
                    </label>
                  </div>
                  <div
                    class="col-8"
                  >
                    <div
                      class="p-inputswitch p-component p-inputswitch-checked p-disabled"
                    >
                      <div
                        class="p-hidden-accessible"
                      >
                        <input
                          aria-checked="true"
                          aria-label="input-checkbox"
                          checked=""
                          disabled=""
                          name="status"
                          type="checkbox"
                        />
                      </div>
                      <span
                        class="p-inputswitch-slider"
                      />
                    </div>
                  </div>
                </div>
                <div
                  class="row"
                >
                  <div
                    class="col-8 offset-4"
                  >
                    <button
                      class="me-2 btn btn-danger"
                      type="button"
                    >
                      common_confirm_cancel
                    </button>
                    <button
                      class="me-2 btn btn-success"
                      disabled=""
                      type="button"
                    >
                      common_confirm_save
                    </button>
                    <button
                      class="me-2 btn btn-primary"
                      type="button"
                    >
                      user_form_button_assigned_resources
                    </button>
                  </div>
                </div>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="card-form mt-3 card"
    >
      <div
        class="card-header"
      >
        <h4
          class="card-form__title"
        >
          user_edit_page_title
        </h4>
      </div>
      <div
        class="card-body"
      >
        <div
          class="row"
        >
          <form
            autocomplete="on"
            class="form-layout"
          >
            <div
              class="container-fluid"
            >
              <div
                class="align-items-center py-1 row"
              >
                <div
                  class="col-4"
                >
                  <label
                    for="id"
                  >
                    Id
                  </label>
                </div>
                <div
                  class="p-fluid col-8"
                >
                  <input
                    aria-label="disabled"
                    class="p-inputtext p-component p-disabled p-1"
                    disabled=""
                    id="id"
                    name="id"
                    value=""
                  />
                </div>
              </div>
              <div
                class="align-items-center py-1 row"
              >
                <div
                  class="col-4"
                >
                  <label
                    for="erpId"
                  >
                    organization_details_erpId
                  </label>
                </div>
                <div
                  class="p-fluid col-8"
                >
                  <input
                    aria-label="disabled"
                    class="p-inputtext p-component p-disabled w-full p-1"
                    disabled=""
                    id="erpId"
                    maxlength="16"
                    name="erpId"
                    value=""
                  />
                </div>
              </div>
              <div
                class="row"
              />
              <div
                class="align-items-center py-1 row"
              >
                <div
                  class="col-4"
                >
                  <label
                    class="required"
                    for="firstName"
                  >
                    user_form_first_name
                  </label>
                </div>
                <div
                  class="p-fluid col-8"
                >
                  <input
                    class="p-inputtext p-component w-full p-1"
                    id="firstName"
                    maxlength="128"
                    name="firstName"
                    value=""
                  />
                </div>
              </div>
              <div
                class="align-items-center py-1 row"
              >
                <div
                  class="col-4"
                >
                  <label
                    class="required"
                    for="lastName"
                  >
                    user_form_last_name
                  </label>
                </div>
                <div
                  class="p-fluid col-8"
                >
                  <input
                    class="p-inputtext p-component w-full p-1"
                    id="lastName"
                    maxlength="128"
                    name="lastName"
                    value=""
                  />
                </div>
              </div>
              <div
                class="align-items-center py-1 row"
              >
                <div
                  class="col-4"
                >
                  <label
                    class="required"
                    for="contactEmailAddress"
                  >
                    user_form_email_address
                  </label>
                </div>
                <div
                  class="p-fluid col-8"
                >
                  <input
                    aria-label="disabled"
                    class="p-inputtext p-component p-disabled w-full p-1"
                    disabled=""
                    id="emailAddress"
                    maxlength="256"
                    name="emailAddress"
                    value=""
                  />
                </div>
              </div>
              <div
                class="row"
              />
              <div
                class="align-items-center py-1 row"
              >
                <div
                  class="col-4"
                >
                  <label
                    class=""
                    for="isActive"
                  >
                    organization_details_status_active
                  </label>
                </div>
                <div
                  class="col-8"
                >
                  <div
                    class="p-inputswitch p-component p-inputswitch-checked p-disabled"
                  >
                    <div
                      class="p-hidden-accessible"
                    >
                      <input
                        aria-checked="true"
                        aria-label="input-checkbox"
                        checked=""
                        disabled=""
                        name="status"
                        type="checkbox"
                      />
                    </div>
                    <span
                      class="p-inputswitch-slider"
                    />
                  </div>
                </div>
              </div>
              <div
                class="row"
              >
                <div
                  class="col-8 offset-4"
                >
                  <button
                    class="me-2 btn btn-danger"
                    type="button"
                  >
                    common_confirm_cancel
                  </button>
                  <button
                    class="me-2 btn btn-success"
                    disabled=""
                    type="button"
                  >
                    common_confirm_save
                  </button>
                  <button
                    class="me-2 btn btn-primary"
                    type="button"
                  >
                    user_form_button_assigned_resources
                  </button>
                </div>
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
